src/main/java/org/generationcp/breeding/manager/customcomponent/DeletePrefixCacheDialog.java

```diff
@@ -0,0 +1,354 @@
+				this.addComponent(component);
+			}
+		}
+	}
+
+	private void instantiateTable() {
+		this.prefixesTable = new ControllableRefreshTable();
+		this.prefixesTable.setWidth("100%");
+		this.prefixesTable.setHeight("300px");
+
+		this.prefixesTable.setData(DeletePrefixCacheDialog.PREFIXES_TABLE_DATA);
+		this.prefixesTable.addContainerProperty(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID, String.class, null);
+		this.prefixesTable.addContainerProperty(DeletePrefixCacheDialog.REMOVE_PROPERTY_ID, Button.class, null);
+
+		this.prefixesTable.setColumnHeader(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID, DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID);
+		this.prefixesTable.setColumnHeader(DeletePrefixCacheDialog.REMOVE_PROPERTY_ID, DeletePrefixCacheDialog.REMOVE_PROPERTY_ID);
+		this.prefixesTable.setColumnWidth(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID, 350);
+		this.prefixesTable.setColumnAlignment(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID, Table.ALIGN_CENTER);
+		this.prefixesTable.setColumnAlignment(DeletePrefixCacheDialog.REMOVE_PROPERTY_ID, Table.ALIGN_CENTER);
+	}
+
+	private void addPrefix() {
+		final String prefix = this.prefixTextField.getValue().toString().trim();
+		if(!StringUtil.isEmpty(prefix)) {
+			if(!this.prefixesTable.getVisibleItemIds().contains(prefix.toUpperCase())) {
+				final Item newItem = this.prefixesTable.getContainerDataSource().addItem(prefix.toUpperCase());
+				newItem.getItemProperty(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID).setValue(prefix);
+
+				final Button removeButton = new Button("REMOVE");
+				removeButton.addListener(new RemovePrefixButtonListener(this));
+				removeButton.setData(prefix.toUpperCase() );
+				newItem.getItemProperty(DeletePrefixCacheDialog.REMOVE_PROPERTY_ID).setValue(removeButton);
+				this.prefixesTable.requestRepaint();
+				this.deletePrefixesButton.setEnabled(true);
+			} else {
+				MessageNotifier
+					.showWarning(this.source.getWindow(), this.messageSource.getMessage(Message.WARNING),
+						this.messageSource.getMessage(Message.WARNING_PREFIX_EXISTING_IN_TABLE));
+			}
+		}
+
+		this.prefixTextField.setValue("");
+	}
+
+	private void deleteFromTable(final String itemId) {
+		this.prefixesTable.getContainerDataSource().removeItem(itemId);
+		this.prefixesTable.requestRepaint();
+		if(CollectionUtils.isEmpty(this.prefixesTable.getVisibleItemIds())) {
+			this.deletePrefixesButton.setEnabled(false);
+		}
+	}
+
+	void deletePrefixes() {
+		final List<String> prefixes = new ArrayList<>((Collection<? extends String>) this.prefixesTable.getVisibleItemIds());
+		final List<String> names = this.germplasmDataManager.getNamesByGidsAndPrefixes(this.deletedGIDs, prefixes);
+		if(!CollectionUtils.isEmpty(names)) {
+			final List<KeySequenceRegister> keySequenceRegistersOfDeletedGermplasm = this.germplasmDataManager.getKeySequenceRegistersByPrefixes(prefixes);
+			if(CollectionUtils.isEmpty(keySequenceRegistersOfDeletedGermplasm)) {
```

Comment:
We don't need to shown an error or warning when there there is no sequence(s) to be deleted.

Reply:
To simplify the logic, we resolve from the user-specified prefixes which ones has regular expression match to retrieved names. Then we use the final list of prefixes to delete from key sequence registry (whether or not they exist).